tests/phpunit/tests/image/functions.php

```diff
@@ -726,6 +726,134 @@ public function test_wp_generate_attachment_metadata_pdf() {
+
+		$metadata = wp_generate_attachment_metadata( $attachment_id, $test_file );
+
+		$expected = array(
+			'sizes'    => array(
+				'full'      => array(
+					'file'      => 'wordpress-gsoc-flyer-pdf.jpg',
+					'width'     => 1088,
+					'height'    => 1408,
+					'mime-type' => 'image/jpeg',
+					'filesize'  => wp_filesize( $temp_dir . 'wordpress-gsoc-flyer-pdf.jpg' ),
+					'sources'   => array(
+						'image/jpeg' => array(
+							'file'     => 'wordpress-gsoc-flyer-pdf.jpg',
+							'filesize' => wp_filesize( $temp_dir . 'wordpress-gsoc-flyer-pdf.jpg' ),
+						),
+						'image/webp' => array(
+							'file'     => 'wordpress-gsoc-flyer-pdf-jpg.webp',
+							'filesize' => wp_filesize( $temp_dir . 'wordpress-gsoc-flyer-pdf-jpg.webp' ),
+						),
+					),
+				),
+				'medium'    => array(
+					'file'      => 'wordpress-gsoc-flyer-pdf-232x300.jpg',
+					'width'     => 232,
+					'height'    => 300,
+					'mime-type' => 'image/jpeg',
+					'filesize'  => wp_filesize( $temp_dir . 'wordpress-gsoc-flyer-pdf-232x300.jpg' ),
+					'sources'   => array(
+						'image/jpeg' => array(
+							'file'     => 'wordpress-gsoc-flyer-pdf-232x300.jpg',
+							'filesize' => wp_filesize( $temp_dir . 'wordpress-gsoc-flyer-pdf-232x300.jpg' ),
+						),
+						'image/webp' => array(
+							'file'     => 'wordpress-gsoc-flyer-pdf-232x300-jpg.webp',
+							'filesize' => wp_filesize( $temp_dir . 'wordpress-gsoc-flyer-pdf-232x300-jpg.webp' ),
+						),
+					),
+				),
+				'large'     => array(
+					'file'      => 'wordpress-gsoc-flyer-pdf-791x1024.jpg',
+					'width'     => 791,
+					'height'    => 1024,
+					'mime-type' => 'image/jpeg',
+					'filesize'  => wp_filesize( $temp_dir . 'wordpress-gsoc-flyer-pdf-791x1024.jpg' ),
+					'sources'   => array(
+						'image/jpeg' => array(
+							'file'     => 'wordpress-gsoc-flyer-pdf-791x1024.jpg',
+							'filesize' => wp_filesize( $temp_dir . 'wordpress-gsoc-flyer-pdf-791x1024.jpg' ),
+						),
+						'image/webp' => array(
+							'file'     => 'wordpress-gsoc-flyer-pdf-791x1024-jpg.webp',
+							'filesize' => wp_filesize( $temp_dir . 'wordpress-gsoc-flyer-pdf-791x1024-jpg.webp' ),
+						),
+					),
+				),
+				'thumbnail' => array(
+					'file'      => 'wordpress-gsoc-flyer-pdf-116x150.jpg',
+					'width'     => 116,
+					'height'    => 150,
+					'mime-type' => 'image/jpeg',
+					'filesize'  => wp_filesize( $temp_dir . 'wordpress-gsoc-flyer-pdf-116x150.jpg' ),
+					'sources'   => array(
+						'image/jpeg' => array(
+							'file'     => 'wordpress-gsoc-flyer-pdf-116x150.jpg',
+							'filesize' => wp_filesize( $temp_dir . 'wordpress-gsoc-flyer-pdf-116x150.jpg' ),
+						),
+						'image/webp' => array(
+							'file'     => 'wordpress-gsoc-flyer-pdf-116x150-jpg.webp',
+							'filesize' => wp_filesize( $temp_dir . 'wordpress-gsoc-flyer-pdf-116x150-jpg.webp' ),
+						),
+					),
+				),
+			),
+			'filesize' => wp_filesize( $test_file ),
+		);
+
+		$this->assertSame( $expected, $metadata );
```

Comment:
Instead of this can we just check if `$metadata` has `image/webp` and check some sizes of the sources file?